app/controllers/api/v1/admin/invitations_controller.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    module Admin
+      class InvitationsController < ApiController
+        before_action only: %i[create] do
+          ensure_authorized('ManageUsers')
+        end
+
+        # POST /api/v1/admin/invitations
+        def create
+          params[:invitations][:emails].split(',').each do |email|
+            invitation = Invitation.find_or_initialize_by(email:, provider: current_provider).tap do |i|
+              i.updated_at = Time.zone.now
+              i.save!
+            end
+
+            UserMailer.with(
+              email:,
+              name: current_user.name,
+              signup_url: root_url(inviteToken: invitation.token)
+            ).invitation_email.deliver_later
+          rescue StandardError => e
+            logger.error "Failed to send invitation to #{email} - #{e}"
+          end
+
+          render_data status: :ok
+        rescue StandardError => e
+          logger.error "Failed to send invitations to #{params[:invitations][:emails]} - #{e}"
```

Comment:
I think this may be hard to follow,
With the current implementation this rescue block seems to be triggered beacuase of `params[:invitations][:emails].split(',').each` which will be probabily caused by an `undefined method 'xxx' for nil:NilClass (NoMethodError)`.
That can easily be validated before proceeding in the logic like we used to do and further clean this.
Besides the same exception can happen here in the resucer and throw it again causing the global rescuer to handle this which seems not to be the case since it removes the need for this,

Reply:
Besides are we willing to treat here a generic exception (almost all running time exceptions) as a bad request and brake from all global rescuing?